src/import/import.rs

```diff
@@ -86,25 +105,119 @@ impl<Client: ImportClient> ImportJob<Client> {
+        range_rx: Receiver<Range>,
+        sst_tx: Sender<LazySSTRange>,
+        finished_ranges: Arc<Mutex<Vec<Range>>>,
+    ) -> JoinHandle<Result<()>> {
+        let engine = Arc::clone(&self.engine);
+        let cfg = self.cfg.clone();
+        let tag = self.tag.clone();
+
+        thread::Builder::new()
+            .name("dispatch-job".to_owned())
+            .spawn(move || {
+                'NEXT_RANGE: while let Ok(range) = range_rx.recv() {
+                    'RETRY: for _ in 0..MAX_RETRY_TIMES {
+                        let cfg = cfg.clone();
+                        let client = Arc::clone(&client);
+                        let engine = Arc::clone(&engine);
+                        let mut stream = SSTFileStream::new(
+                            cfg,
+                            client,
+                            engine,
+                            range.clone(),
+                            finished_ranges.lock().unwrap().clone(),
+                        );
+
+                        loop {
+                            let split_start = Instant::now_coarse();
+                            match stream.next() {
+                                Ok(Some(info)) => {
+                                    IMPORT_SPLIT_SST_DURATION.observe(split_start.elapsed_secs());
+                                    let start = Instant::now_coarse();
+                                    sst_tx.send(info).unwrap();
+                                    IMPORT_SST_DELIVERY_DURATION.observe(start.elapsed_secs());
+                                }
+                                Ok(None) => continue 'NEXT_RANGE,
+                                Err(_) => continue 'RETRY,
+                            }
+                        }
+                    }
+                }
+
+                info!("dispatch-job done"; "tag" => %tag, "id" => %id);
+                Ok(())
+            })
+            .unwrap()
+    }
+
+    fn run_import_threads(
+        &self,
+        ranges: Vec<Range>,
+        retry_ranges: Arc<Mutex<Vec<Range>>>,
+    ) -> Vec<JoinHandle<Result<()>>> {
+        let mut handles: Vec<JoinHandle<Result<()>>> = Vec::new();
+        let finished_ranges = Arc::new(Mutex::new(Vec::new()));
+        let (sst_tx, sst_rx) = bounded(self.cfg.num_import_jobs);
+
+        // Spawn a group of threads to execute real import jobs
+        for id in 0..self.cfg.num_import_jobs {
+            handles.push(self.new_import_thread(
+                id as u64,
+                sst_rx.clone(),
+                Arc::clone(&retry_ranges),
+            ));
+        }
+
+        // Spawn a thread to dispatch ranges
+        let split_thread_count = self.cfg.num_import_jobs * 2;
+        let (range_tx, range_rx) = bounded(split_thread_count);
+        let range_handle = thread::Builder::new()
+            .name("dispatch-range-job".to_owned())
+            .spawn(move || {
+                for (_, range) in ranges.into_iter().enumerate() {
```

Comment:
why need a thread to do this?

Reply:
Yes, we need a dispatcher thread to dispatch range to split SST threads.
In the before implemetation, we will spawn a large of threads to split SST file when the Engine file is huge.

Reply:
But seems it isn't an expensive operation if using `unbound` channel, why not do it just in this thread instead of spawning a new thread.

Reply:
addressed